iothub/service/samples/solutions/PnpServiceSamples/TemperatureController/TemperatureControllerSample.cs

```diff
@@ -122,8 +119,9 @@ private async Task UpdateDeviceTwinComponentPropertyAsync()
             var twinPatch = CreatePropertyPatch(targetTemperaturePropertyName, desiredTargetTemperature, Thermostat1Component);
             _logger.LogDebug($"Updating the {targetTemperaturePropertyName} property under component {Thermostat1Component} on the {_deviceId} device twin to { desiredTargetTemperature}.");
 
-            Twin twin = await _registryManager.GetTwinAsync(_deviceId);
-            await _registryManager.UpdateTwinAsync(_deviceId, twinPatch, twin.ETag);
+            Twin currentTwin = await _serviceClient.Twins.GetAsync(_deviceId);
+            twinPatch.ETag = currentTwin.ETag;
```

Comment:
ETag is now part of the patch, and the bool parameter indicates if one cares about respecting the ETag freshness.